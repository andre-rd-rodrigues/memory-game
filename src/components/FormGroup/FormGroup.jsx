import React, { useEffect, useState, memo } from "react";
import styles from "./formgroup.module.scss";
import colors from "styles/_mixins.scss";

const FormGroup = ({ label, inputs = [""], onChange }) => {
  const [selected, setSelected] = useState(inputs[0]);

  const renderStyle = (button) => {
    return {
      backgroundColor: selected === button ? colors.primary : ""
    };
  };

  useEffect(() => {
    onChange(selected);
  }, [selected]);

  return (
    <div className={styles.formGroup}>
      <p className="form-label">{label}</p>
      <div className="form-select">
        {inputs.map((button, index) => (
          <button
            key={index}
            onClick={() => setSelected(button)}
            style={renderStyle(button)}
          >
            {button}
          </button>
        ))}
      </div>
    </div>
  );
};

export default memo(FormGroup);
