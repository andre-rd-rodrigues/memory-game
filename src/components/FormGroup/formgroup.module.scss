@import "styles/mixins";

.formGroup {
  margin-bottom: 1.5rem;
  & p {
    color: $primary-light;
    font-size: 0.9rem;
  }
  & div[class="form-select"] {
    display: flex;
    & button {
      @include button($secondary, white);
      margin: 0 0.5rem;
      width: 100%;
    }
  }
}
