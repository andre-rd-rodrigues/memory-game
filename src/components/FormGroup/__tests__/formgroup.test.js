import { shallow } from "enzyme";
import FormGroup from "../FormGroup";

describe("Form Group", () => {
  let props;
  props = {
    inputs: ["Icons", "Numbers"],
    label: "Select Theme",
    onChange: jest.fn()
  };

  it("should render correctly", () => {
    const wrapper = shallow(<FormGroup {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
