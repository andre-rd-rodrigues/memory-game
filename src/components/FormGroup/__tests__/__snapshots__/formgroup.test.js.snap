// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Group should render correctly 1`] = `
<div
  className="formGroup"
>
  <p
    className="form-label"
  >
    Select Theme
  </p>
  <div
    className="form-select"
  >
    <button
      key="0"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      Icons
    </button>
    <button
      key="1"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
        }
      }
    >
      Numbers
    </button>
  </div>
</div>
`;
